### PR TITLE
Added GTC-OSIRIS+ to fits2hcm

### PR DESCRIPTION
**Pull request by Alex-J-Brown**
I've added OSIRIS+ to the fits2hcm script. I've only tested this with the standard 2x2 and full frame broadband imaging setup. Note that OSIRIS+ is different to the previous version of OSIRIS with a different detector and I believe different FITS header keys.

### CODE REVIEW

File: hipercam/scripts/fits2hcm.py [hunk 1]
```diff
@@ -65,14 +65,20 @@ def fits2hcm(args=None):
            WHTCAM :
              Camera used by Richard Ashley in Feb 2021.
 
+           OSIRIS :
+             Imager and spectrograph on the GTC. I've only tested this with
+             data taken with the new detector known as OSIRIS+ and even then,
+             only with what I believe is the standard setup
+             (2x2 binning and full frame).
+
```
Comment (StuartLittlefair) [inline hipercam/scripts/fits2hcm.py:73]: Can we change this docstring to assign blame, and also the name of the instrument? So, something like

```
OSIRIS+ : 
     Imager and spectrograph on the GTC. Added by Alex Brown in Nov 2023. 
     Only tested with 2x2 binning and full frame
```

File: hipercam/scripts/fits2hcm.py [hunk 2]
```diff
@@ -65,14 +65,20 @@ def fits2hcm(args=None):
...
       overwrite : bool
          overwrite files on output
 
     """
 
     command, args = cline.script_args(args)
 
-    FORMATS = ["HICKS", "INTWFC", "LTRISE", "LTIO", "PT5M", "ROSA", "LCOGT", "WHTCAM"]
+    FORMATS = ["HICKS", "INTWFC", "LTRISE", "LTIO", "PT5M", "ROSA", "LCOGT", "WHTCAM", "OSIRIS"]
```
Comment (StuartLittlefair) [inline hipercam/scripts/fits2hcm.py:81]: should be `OSIRIS+`

File: hipercam/scripts/fits2hcm.py [hunk 3]
```diff
@@ -499,6 +505,44 @@ def fits2hcm(args=None):
                     ohdul = fits.HDUList([ophdu, ofhdu])
                     ohdul.writeto(oname, overwrite=overwrite)
 
+                elif origin == "OSIRIS":
```
Comment (StuartLittlefair) [inline hipercam/scripts/fits2hcm.py:508]: `OSIRIS+`

File: hipercam/scripts/fits2hcm.py [hunk 4]
```diff
@@ -499,6 +505,44 @@ def fits2hcm(args=None):
...
                     ohdul.writeto(oname, overwrite=overwrite)
 
+                elif origin == "OSIRIS":
+                    # Copy main header into primary data-less HDU
+                    ihead = hdul[0].header
+                    ophdu = fits.PrimaryHDU(header=ihead)
+                    ophdu.header["NUMCCD"] = (1, "CCD number; fits2hcm")
+                    exptime = ihead["EXPTIME"]
+                    mjd = ihead["MJD-OBS"] + exptime / 2 / 86400
+                    time = Time(mjd + exptime / 2 / 86400, format="mjd")
```
Comment (StuartLittlefair) [inline hipercam/scripts/fits2hcm.py:515]: You're adding half the exposure time twice here, I believe.

File: hipercam/scripts/fits2hcm.py [hunk 5]
```diff
@@ -499,6 +505,44 @@ def fits2hcm(args=None):
...
+                    ophdu.header["TIMSTAMP"] = (time.isot, "Time stamp; fits2hcm")
+
+                    ofhdu = fits.ImageHDU(hdul[0].data)
+
+                    ofhdu.header["CCD"] = ("1", "CCD label")
+                    ofhdu.header["NXTOT"] = (4096, "Total unbinned X dimension")
+                    ofhdu.header["NYTOT"] = (4112, "Total unbinned Y dimension")
+                    ofhdu.header["NUMWIN"] = (ihead['NUM_ROIS'], "Total number of windows")
+                    ofhdu.header["WINDOW"] = ("1", "Window label")
+                    ofhdu.header["LLX"] = (1, "X-ordinate of lower-left pixel")
```
Comment (StuartLittlefair) [inline hipercam/scripts/fits2hcm.py:525]: It would be good to get our hands on some windowed data to add this, but OK for now

**Review by StuartLittlefair — CHANGES_REQUESTED** [review]

**Comment by Alex-J-Brown** [comment]
I've made the requested changes but noticed that the extra half exposure time error was present in the code for a lot of the other instruments so I've removed those too.

**Review by StuartLittlefair — APPROVED** [review]

**Pull request merged**